administration/backpressure.md

```diff
@@ -2,68 +2,119 @@
+memory reaches this limit and you reach a backpressure scenario, you won't be able
+to ingest more data until the data chunks that are in memory can be flushed. The
+input pauses and Fluent Bit
+[emits](https://github.com/fluent/fluent-bit/blob/v2.0.0/src/flb_input_chunk.c#L1334)
+a `[warn] [input] {input name or alias} paused (mem buf overlimit)` log message.
+
+Depending on the input plugin in use, this might lead to discard incoming data (for
+example, TCP input plugin). The tail plugin can handle pause without data loss; it
+stores its current file offset and resumes reading later. When buffer memory is
+available, the input resumes accepting logs. Fluent Bit
+[emits](https://github.com/fluent/fluent-bit/blob/v2.0.0/src/flb_input_chunk.c#L1277)
+a `[info] [input] {input name or alias} resume (mem buf overlimit)` message.
+
+Mitigate the risk of data loss by configuring secondary storage on the filesystem
+using the `storage.type` of `filesystem` (as described in [Buffering &
+Storage](buffering-and-storage.md)). Initially, logs will be buffered to both memory
+and the filesystem. When the `storage.max_chunks_up` limit is reached, all new data
+will be stored safely in the filesystem. Fluent Bit stops queueing new data in memory
+and will only buffers to the filesystem. When `storage.type filesystem` is set, the
+`Mem_Buf_Limit` setting no longer has any effect. Instead, the `[SERVICE]` level
+`storage.max_chunks_up` setting controls the size of the memory buffer.
+
+## `Mem_Buf_Limit`
+
+`Mem_Buf_Limit` only applies with the default `storage.type memory`. This option is
+disabled by default and can be applied to all input plugins.
+
+As an example situation:
+
+- `Mem_Buf_Limit` is set to `1MB`.
+- The input plugin tries to append 700KB.
+- The engine routes the data to an output plugin.
+- The output plugin backend (HTTP Server) is down.
+- Engine scheduler retries the flush after 10 seconds.
+- The input plugin tries to append 500KB.
+
+In this situation, the engine allows appending those 500KB of data into the memory,
+with a total of 1.2 MB of data buffered. The limit is permissive and will
+allow a single write past the limit. Once the limit is exceeded, the following
+actions are taken:
+
+- Block local buffers for the input plugin (can't append more data).
+- Notify the input plugin, invoking a `pause` callback.
+
+The engine protects itself and won't append more data coming from the input plugin in
+question. It's the responsibility of the plugin to keep state and decide what to do
+in a `paused` state.
+
+In a few seconds, if the scheduler was able to flush the initial 700KB of data or it
+has given up after retrying, that amount of memory is released and the following
+actions occur:
+
+- Upon data buffer release (700KB), the internal counters get updated.
+- Counters now are set at 500KB.
+- Since 500KB is &lt; 1 MB it checks the input plugin state.
```

Comment:
```suggestion
- Upon data buffer release (700&nbsp;KB), the internal counters get updated.
- Counters now are set at 500&nbsp;KB.
- Because 500&nbsp;KB isless than 1&nbsp;MB, it checks the input plugin state.
```